include/photos.php

```diff
@@ -42,16 +45,28 @@ function photo_albums($uid, $update = false) {
 	$key = "photo_albums:".$uid.":".local_user().":".remote_user();
 	$albums = Cache::get($key);
 	if (is_null($albums) OR $update) {
-		/// @todo This query needs to be renewed. It is really slow
-		// At this time we just store the data in the cache
-		$albums = qu("SELECT COUNT(DISTINCT `resource-id`) AS `total`, `album`
-			FROM `photo` USE INDEX (`uid_album_created`)
-			WHERE `uid` = %d  AND `album` != '%s' AND `album` != '%s' $sql_extra
-			GROUP BY `album` ORDER BY `created` DESC",
-			intval($uid),
-			dbesc('Contact Photos'),
-			dbesc(t('Contact Photos'))
-		);
+		if (!Config::get('system', 'no_count', false)) {
+			/// @todo This query needs to be renewed. It is really slow
```

Comment:
Yes, best thing is to make all cache entries invalid in the moment when there is some change in the photos or albums. Question is only how to do it with the current cache system. We would need some wildcard cache deletion system.